### PR TITLE
fixed bug in run_mlm_flax_stream.py

### LINKED ISSUE #17132: Dataset streaming example not working
### System Info

```shell
- `transformers` version: 4.18.0
- Platform: Linux-5.4.173.el7-x86_64-with-glibc2.10
- Python version: 3.8.12
- Huggingface_hub version: 0.5.1
- PyTorch version (GPU?): 1.11.0a0+17540c5 (True)
- Tensorflow version (GPU?): 2.8.0 (True)
- Flax version (CPU?/GPU?/TPU?): 0.4.2 (gpu)
- Jax version: 0.3.10
- JaxLib version: 0.3.10
- Using GPU in script?: Yes
- Using distributed or parallel set-up in script?: No
```

### Who can help?

@patrickvonplaten 

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [X] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

Following the guide to train a model in streaming mode using the [dataset-streaming](https://github.com/huggingface/transformers/tree/main/examples/research_projects/jax-projects/dataset-streaming) directory results in the following error.

```
[11:11:16] - INFO - datasets_modules.datasets.oscar.84838bd49d2295f62008383b05620571535451d84545037bb94d6f3501651df2.oscar - generating examples from = https://s3.amazonaws.com/datasets.huggingface.co/oscar/1.0/unshuffled/deduplicated/en/en_part_480.txt.gz
Token indices sequence length is longer than the specified maximum sequence length for this model (1195 > 512). Running this sequence through the model will result in indexing errors
Traceback (most recent call last):
  File "./run_mlm_flax_stream.py", line 549, in <module>
    eval_samples = advance_iter_and_group_samples(training_iter, data_args.num_eval_samples, max_seq_length)
  File "./run_mlm_flax_stream.py", line 284, in advance_iter_and_group_samples
    samples = {k: samples[k] + tokenized_samples[k] for k in tokenized_samples.keys()}
  File "./run_mlm_flax_stream.py", line 284, in <dictcomp>
    samples = {k: samples[k] + tokenized_samples[k] for k in tokenized_samples.keys()}
TypeError: can only concatenate list (not "int") to list
```

### Expected behavior

```shell
Model training to start.
```

### PR DESCRIPTION
**Pull request by KennethEnevoldsen**
# What does this PR do?

Fixed bug caused by additional keys (id, text) referring to non-list values when concatenating samples. An alternative option is to drop these columns in the dataset before passing it to `iter`.

Fixes #17132 by @HLasse.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@patrickvonplaten (tagged in bug)
@patil-suraj (assigned to bug)

### CODE REVIEW

**Review by patil-suraj — COMMENTED** [review]
Thanks for the PR. This solution looks good, but I think the better option would to just drop the other columns, like its done in PT script.

https://github.com/huggingface/transformers/blob/a42242da7c44d64c66a878cca65bc86dd3f626af/examples/pytorch/language-modeling/run_mlm.py#L379

https://github.com/huggingface/transformers/blob/a42242da7c44d64c66a878cca65bc86dd3f626af/examples/pytorch/language-modeling/run_mlm.py#L440

**Comment by KennethEnevoldsen** [comment]
No problem. Tried using the `remove_columns ` argument as in the example but the map functions do not take that argument (probably easy to add):

```python
tokenized_datasets = dataset.map(
        tokenize_function,
        batched=True,
        remove_columns=column_names,
    )
```

Similar it might be nice for consistency to add the `.column_names` extensions to the IterableDataset.

File: examples/research_projects/jax-projects/dataset-streaming/run_mlm_flax_stream.py [hunk 1]
```diff
@@ -401,10 +401,12 @@ def write_eval_metric(summary_writer, eval_metrics, step):
     def tokenize_function(examples):
         return tokenizer(examples[data_args.text_column_name], return_special_tokens_mask=True)
 
+
     tokenized_datasets = dataset.map(
         tokenize_function,
         batched=True,
     )
+    tokenized_datasets.remove_columns(dataset.features.keys())
```
Comment (patil-suraj) [inline examples/research_projects/jax-projects/dataset-streaming/run_mlm_flax_stream.py:408]: any reason not to pass the `remove_columns` argument to `dataset.map` ? Like 

```python3
tokenized_datasets = dataset.map(
        tokenize_function,
        batched=True,
        remove_columns=dataset["train"].column_names
)
```

Comment (patil-suraj) [reply examples/research_projects/jax-projects/dataset-streaming/run_mlm_flax_stream.py:408]: Aah, just read your comment above. Ignore this.

**Comment by patil-suraj** [comment]
> No problem. Tried using the `remove_columns ` argument as in the example but the map functions do not take that argument (probably easy to add):
> 
> ```python
> tokenized_datasets = dataset.map(
>         tokenize_function,
>         batched=True,
>         remove_columns=column_names,
>     )
> ```
> 
> Similar it might be nice for consistency to add the `.column_names` extensions to the IterableDataset.

Pinging @lhoestq here. Is not possible to pass remove column to streaming dataset ?

**Comment by lhoestq** [comment]
`remove_columns` does exist: 

https://huggingface.co/docs/datasets/v2.2.1/en/package_reference/main_classes#datasets.IterableDataset.map.remove_columns

**Comment by KennethEnevoldsen** [comment]
Indeed it does. I was using datasets version `1.17.0`. Rerunning with an updated version indeed resolved the issue.

**Comment by KennethEnevoldsen** [comment]
Another note here, seems like the max_seq_length state that each text is truncated to e.g. max_seq_length, but from what I can read in the `advance_iter_and_group_samples` it seems like they are concatenated and split in groups of max_seq_length (with the remainder being discarded)

I would probably change it from:
```python
    max_seq_length: Optional[int] = field(
        default=None,
        metadata={
            "help": "The maximum total input sequence length after tokenization. Sequences longer "
            "than this will be truncated. Default to the max input length of the model."
        },
    )
```
to:
```python
    max_seq_length: Optional[int] = field(
        default=None,
        metadata={
            "help": "The maximum total input sequence length after tokenization. Sequences are concatenated in groups of max_seq_length. Default to the max input length of the model."
        },
    )
```

**Review by patil-suraj — APPROVED** [review]
LGTM!

**Comment by patil-suraj** [comment]
> Another note here, seems like the max_seq_length state that each text is truncated to e.g. max_seq_length, but from what I can read in the `advance_iter_and_group_samples` it seems like they are concatenated and split in groups of max_seq_length (with the remainder being discarded)
> 
> I would probably change it from:
> 
> ```python
>     max_seq_length: Optional[int] = field(
>         default=None,
>         metadata={
>             "help": "The maximum total input sequence length after tokenization. Sequences longer "
>             "than this will be truncated. Default to the max input length of the model."
>         },
>     )
> ```
> 
> to:
> 
> ```python
>     max_seq_length: Optional[int] = field(
>         default=None,
>         metadata={
>             "help": "The maximum total input sequence length after tokenization. Sequences are concatenated in groups of max_seq_length. Default to the max input length of the model."
>         },
>     )
> ```

Good catch! Feel free to open another PR if you would like to fix this.

**Pull request merged**